Assets/MixedRealityToolkit.SDK/Features/UX/Scripts/Utilities/UIMaterialInstance.cs

```diff
@@ -0,0 +1,54 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+using System.Collections.Generic;
+using UnityEngine;
+using UnityEngine.UI;
+
+namespace Microsoft.MixedReality.Toolkit.UI
+{
+    public class UIMaterialInstance
+    {
+        // this set ensures that we do not end up creating multiple copies of materials for every theme targetting the same instance
+        private static HashSet<int> targetInstances = new HashSet<int>();
+
+        private Material material = null;
+        private Graphic targetGraphic;
+
+        public UIMaterialInstance(Graphic targetObject)
+        {
+            targetGraphic = targetObject;
+            material = targetGraphic.material;
+        }
+
+        public void TryCreateMaterialCopy()
```

Comment:
Should this just be thrown in the constructor? If I am deliberately creating a UIMaterialInstance, I feel like I should automatically get an instanced material. 

Suggestion: Simplify this class as a static helper. Get rid of the get/set color etc. Those aren't doing a lot that I as a consumer can't do myself on the material. Further, there are all of the setvector or other random primitive types not covered here

Instead of the constructor, I would just call static TryGetMaterialInstance() and then cache that in my theme or other use case